### PR TITLE
Refactor `databricks_permissions` to common.Resource

### PR DESCRIPTION
**Pull request by nfx**
This guarantees panic recovery and common error messages across all resources

### CODE REVIEW

File: access/resource_ipaccesslist_test.go [hunk 1]
```diff
@@ -313,25 +313,23 @@ func TestIPACLDelete(t *testing.T) {
...
 		},
 		Resource: ResourceIPAccessList(),
 		Delete:   true,
+		Removed:  true,
 		ID:       TestingID,
-	}.Apply(t)
-	assert.Error(t, err)
-	qa.AssertErrorStartsWith(t, err, "IP access list is not available in ")
-	assert.Equal(t, TestingID, d.Id())
+	}.ApplyNoError(t)
```
Comment (pietern) [inline access/resource_ipaccesslist_test.go:332]: Test name indicates it expects an error but here you expect no error.

File: permissions/resource_permissions.go [hunk 2]
```diff
@@ -336,37 +336,7 @@ func ResourcePermissions() *schema.Resource {
...
 		return s
 	})
-	readContext := func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-		id := d.Id()
-		objectACL, err := NewPermissionsAPI(ctx, m).Read(id)
-		if common.IsMissing(err) {
-			log.Printf("[INFO] %s is removed on backend", d.Id())
-			d.SetId("")
-			return nil
-		}
```
Comment (pietern) [inline permissions/resource_permissions.go:346]: This is missing from the new `Read` implementation. Could it regress?

Comment (nfx) [reply permissions/resource_permissions.go:346]: this is part of `common.Resource`, which already consistently handles it across all resources

Comment (nfx) [reply permissions/resource_permissions.go:346]: covered by `TestResourcePermissionsRead_NotFound`

File: access/resource_ipaccesslist_test.go (hunk 1 above)
Comment (nfx) [reply access/resource_ipaccesslist_test.go:332]: fixed in the next commit

**Review by pietern — APPROVED** [review]

**Pull request merged**